src/toil/jobStores/abstractJobStore.py

```diff
@@ -933,10 +935,7 @@ def _supportsUrl(cls, url, export=False):
 
     @classmethod
     def _readFromUrl(cls, url, writable):
-        out = urllib2.urlopen(url.geturl())
-        while True:
-            toWrite = out.read(2**20)
-            if toWrite in ('', None):
-                break
-            else:
-                writable.write(toWrite)
+        for attempt in retry_http():
+            with attempt:
+                with closing(urllib2.urlopen(url.geturl())) as urlF:
```

Comment:
Rename urlF to readable like everywhere else. Please stop inventing new naming conventions.